bin/describe/aliases_targets.ml

```diff
@@ -104,33 +108,56 @@ module Aliases_cmd = struct
     in
     List.map ~f:Dune_engine.Alias.Name.to_string alias_targets
 
-  let term = ls_term fetch_results
+  let term =
+    let+ common, paths, context = common_args in
+    ls_term ~common ~paths ~context fetch_results
 
   let command =
     let doc = "Print aliases in a given directory. Works similalry to ls." in
     Cmd.v (Cmd.info "aliases" ~doc ~envs:Common.envs) term
 end
 
 module Targets_cmd = struct
-  let fetch_results (dir : Path.Build.t) =
+  let fetch_results ~all (dir : Path.Build.t) =
     let open Action_builder.O in
-    let+ targets =
+    let* targets =
       let open Memo.O in
       Target.all_direct_targets (Some (Path.Build.drop_build_context_exn dir))
       >>| Path.Build.Map.to_list |> Action_builder.of_memo
     in
-    List.filter_map targets ~f:(fun (path, kind) ->
+    Action_builder.List.filter_map targets ~f:(fun (path, kind) ->
         match Path.Build.equal (Path.Build.parent_exn path) dir with
-        | false -> None
-        | true ->
-          (* directory targets can be distinguied by the trailing path seperator
-          *)
-          Some
-            (match kind with
-            | Target.File -> Path.Build.basename path
-            | Directory -> Path.Build.basename path ^ Filename.dir_sep))
+        | false -> Action_builder.return None
+        | true -> (
+          (* directory targets can be distinguished by the trailing path
+             seperator. *)
+          match kind with
+          | Target.File ->
+            let+ exists =
+              Dune_engine.Fs_memo.file_exists
+                (Path.as_outside_build_dir_exn
+                   (Path.source (Path.Build.drop_build_context_exn path)))
+              |> Action_builder.of_memo
+            in
+            (* Check if the file exists in the source tree. If it does, we
+               skip printing this target unless the `--all` option was passed *)
+            if (not all) && exists then None
+            else Some (Path.Build.basename path)
+          | Directory ->
+            Action_builder.return
+              (Some (Path.Build.basename path ^ Filename.dir_sep))))
 
-  let term = ls_term fetch_results
+  let term =
+    let+ common, paths, context = common_args
+    and+ all =
+      Arg.(
+        value & flag
+        & info [ "all" ]
+            ~doc:
+              "Print all targets, including those that exist in the source \
+               tree.")
```

Comment:
"All targets including those that exist in the source tree" doesn't make much sense in either of the two worlds:
* If one considers sources to be targets, then the "including ..." part seems tautological/unnecessary.
* If one doesn't consider sources to be targets, then the "print all targets" makes no sense.